src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_cipher.c

```diff
@@ -0,0 +1,363 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_cipher.h"
+#include "pal_utilities.h"
+
+// just some unique IDs
```

Comment:
This comment leads me to believe the values below are arbitrary. If they are indeed arbitrary then I think it might make more sense to create a table for indexing into. The strings and the lengths can all be encoded into a singular place and would be substantially faster.

Reply:
Yes, the values are completely arbitrary.

I'll make this a table that the exposed functions can index into.